### PR TITLE
[ci] Enable Rails/SaveBang Rubocop cop

### PR DESCRIPTION
**Pull request by Ana06**

### CODE REVIEW

**Comment by Ana06** [comment]
[Enable Rails/SaveBang Rubocop cop](https://trello.com/c/OqnDLGSY/284-enable-rails-savebang-rubocop-cop)

**Comment by adrianschroeter** [comment]
the second commit has many code changes, it is too much for me to review it, but please don't tag such commits with "[ci]" only. It should only be used when there is no effect on production can happen.

**Comment by Ana06** [comment]
@adrianschroeter 

> the second commit has many code changes, it is too much for me to review it, but please don't tag such commits with "[ci]" only. It should only be used when there is no effect on production can happen.

Ok, I'll change the tag. The offenses are automatically corrected with Rubocop.

File: src/api/app/models/bs_request.rb [hunk 1]
```diff
@@ -345,14 +345,14 @@ def obsolete_reviews(opts)
...
+        review.save!
         history = HistoryElement::ReviewObsoleted
-        history.create(review: review, comment: "reviewer got removed", user_id: User.current.id)
+        history.create!(review: review, comment: "reviewer got removed", user_id: User.current.id)
 
         # Maybe this will turn the request into an approved state?
         if state == :review && reviews.where(state: "new").none?
           self.state = :new
-          save
+          save!
```
Comment (DavidKang) [inline src/api/app/models/bs_request.rb]: @Ana06, this `save!` method is not a common `save!` of rails, it has been overridden in line 64

File: src/api/app/models/bs_request.rb [hunk 2]
```diff
@@ -374,7 +374,7 @@ def remove_from_group(group)
     end
     self.comment = "removed from group #{group.bs_request.number}"
     self.state = :new
-    save
+    save!
```
Comment (DavidKang) [inline src/api/app/models/bs_request.rb]: The same here

**Comment by adrianschroeter** [comment]
On Montag, 20. März 2017, 02:08:18 CET wrote Ana María Martínez Gómez:
> @adrianschroeter 
> 
> > the second commit has many code changes, it is too much for me to review it, but please don't tag such commits with "[ci]" only. It should only be used when there is no effect on production can happen.
> 
> Ok, I'll change the tag. The offenses are automatically corrected with Rubocop.

Urgs, but you have reviewed the changes?

(we had some breakages due to rubocop "corrections" these days...)

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

**Comment by DavidKang** [comment]
@Ana06, rebase please 😸

**Comment by Ana06** [comment]
@adrianschroeter 

> Urgs, but you have reviewed the changes?

Yes, the problem is that sometimes we overwrite the Rails default methods (as @DavidKang mention), and that make it difficult to review it. I don't know, what about doing it in small steps, starting by  the controllers of the models fro example?

**Pull request closed without merging**